### PR TITLE
feat(safe-apps): Add some e2e tests for the Safe Apps section

### PR DESCRIPTION
**Pull request by yagopv**
## What it solves
We are adding some basic e2e tests ensuring Safe Apps main features are working properly

### CODE REVIEW

File: .github/workflows/safe-apps-e2e.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        run: yarn serve &
+
+      - uses: cypress-io/github-action@v4
+        with:
+          spec: cypress/e2e/safe-apps/*.cy.js
+          browser: chrome
+          record: true
+          config: baseUrl=http://localhost:8080
+        env:
+          CYPRESS_PROJECT_ID: okn21k
```
Comment (yagopv) [inline .github/workflows/safe-apps-e2e.yml:60]: @dasanra we can avoid changing dynamically the cypress.config.js by adding our projectId here. I think is not important as we have public dashboards but we can create a secret if you prefer that

File: cypress/e2e/safe-apps/apps_list.cy.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import { TEST_SAFE } from './constants'
+
+describe('The Safe Apps list', () => {
+  before(() => {
+    cy.visit(`/${TEST_SAFE}/apps`, { failOnStatusCode: false })
+    cy.findByText(/accept selection/i).click()
+  })
+
+  describe('When searching apps', () => {
```
[the remark this answers is not in the document]
Comment (yagopv) [reply cypress/e2e/safe-apps/apps_list.cy.js:9]: oops, let me revisit it

File: cypress/e2e/safe-apps/browser_permissions.cy.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+describe('The Browser permissions system', () => {
+  describe('When the safe app requires permissions', () => {
+    beforeEach(() => {
+      cy.fixture('safe-app').then((html) => {
+        cy.intercept('GET', `${appUrl}/*`, html)
+        cy.intercept('GET', `*/manifest.json`, {
+          name: 'Cypress Test App',
+          description: 'Cypress Test App Description',
+          iconPath: 'http://via.placeholder.com/32',
```
[the remark this answers is not in the document]
Comment (yagopv) [reply cypress/e2e/safe-apps/browser_permissions.cy.js:13]: changed

File: cypress/e2e/safe-apps/apps_list.cy.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (yagopv) [reply cypress/e2e/safe-apps/apps_list.cy.js:9]: I'm testing the search returns a card here so if it does not work is going to fail right? ... Not sure what you mean 😭

[the remark this answers is not in the document]
Comment (mikhailxyz) [reply cypress/e2e/safe-apps/apps_list.cy.js:9]: Yes, but what if the search doesn't actually search and just leaves the whole app list in its initial state?

[the remark this answers is not in the document]
Comment (yagopv) [reply cypress/e2e/safe-apps/apps_list.cy.js:9]: ok, so if the user enter something and the result is list.length ... Ok I can add it

**Review by mikhailxyz — APPROVED** [review]

**Pull request merged**